### DISCUSSION TITLE
如何用adb shell如何启动clash meta并自动打开代理呢

### DISCUSSION THREAD

**Discussion by mailfarmer** [MetaCubeX/ClashMetaForAndroid#353] [Q&A]
手机上装了macrodroid，想实现如下功能，连上家里WiFi，自动关闭clash meta，断开家里Wifi，自动打开clash meta并使能代理。
我在macrodroid的模板市场找到了类似的模板，它是调用adb shell来实现关闭与启动clash meta的，关闭没有问题，启动clash meta也可以，就是启动后不能自动使能代理，需要人工去点击启动。
启动用的shell脚本如下：
adb shell am start-foreground-service com.github.metacubex.clash.meta/com.github.kr328.clash.service.TunService

请问我该如何做

**Comment by xishang0128** [comment]
adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.START_CLASH


adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.STOP_CLASH

**Reply by mailfarmer** [reply]
感谢感谢，脚本已经能正常启用了

**Reply by ghzxs** [reply]
> adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.START_CLASH
> 
> adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.STOP_CLASH

![Screenshot_20241017-151052](https://github.com/user-attachments/assets/0caae84a-fe39-47bc-9c18-0c23fa3cec65)
你好，你是开发者吗？在使用停止命令遇到一些错误，执行停止命令不能正常停止服务，这种情况发生在最近任务没有cmfa窗口

**Reply by yjf2614** [reply]
> > adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.START_CLASH
> > adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.STOP_CLASH
> 
> ![Screenshot_20241017-151052](https://private-user-images.githubusercontent.com/54847264/377351702-0caae84a-fe39-47bc-9c18-0c23fa3cec65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwNzk1NzAsIm5iZiI6MTczMDA3OTI3MCwicGF0aCI6Ii81NDg0NzI2NC8zNzczNTE3MDItMGNhYWU4NGEtZmUzOS00N2JjLTljMTgtMGMyM2ZhM2NlYzY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI4VDAxMzQzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiNTM4ODM3ZDc2M2I5NGI2Y2E2NTMxODM5YTYxMTQ0MjEyMDQ1ZjdjMDVkNWZkZGUzNzFmZjM1OGI4YTNjMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MDvlgc9A_E9aD_uF_o5L2Lsgn2tvFcmYNEz-d3vvKDo) 你好，你是开发者吗？在使用停止命令遇到一些错误，执行停止命令不能正常停止服务，这种情况发生在最近任务没有cmfa窗口

我使用taksker，同样问题，Start命令正常，STOP命令提示如楼上，实际未停止。
目前使用强制关闭clash程序。

**Reply by JoffreyN** [reply]
请问com.github.kr328.clash这个应用也支持这样的操作吗？命令是怎样的？

**Reply by FtyLollipop** [reply]
> > adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.START_CLASH
> > adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.STOP_CLASH
> 
> ![Screenshot_20241017-151052](https://private-user-images.githubusercontent.com/54847264/377351702-0caae84a-fe39-47bc-9c18-0c23fa3cec65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczMzY0MDksIm5iZiI6MTczNzMzNjEwOSwicGF0aCI6Ii81NDg0NzI2NC8zNzczNTE3MDItMGNhYWU4NGEtZmUzOS00N2JjLTljMTgtMGMyM2ZhM2NlYzY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIwVDAxMjE0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5YzkwNTJjOGRlN2RhOTU5MDg0MTYzYWMxMDljMGM5ZmU5YTNhNzgxZWQyNjMwMWE0MDRmNDE4YTFlZGIxZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1v_0VJI0ryNZIez-FDqA7CAd7_asN0Smwk9xnwmxPd4) 你好，你是开发者吗？在使用停止命令遇到一些错误，执行停止命令不能正常停止服务，这种情况发生在最近任务没有cmfa窗口

看来跟我遇到的问题差不多啊
#421

**Reply by DAWN-darkkkkk** [reply]
> > adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.START_CLASH
> > adb shell am start -n com.github.metacubex.clash.meta/com.github.kr328.clash.ExternalControlActivity -a com.github.metacubex.clash.meta.action.STOP_CLASH
> 
> ![Screenshot_20241017-151052](https://private-user-images.githubusercontent.com/54847264/377351702-0caae84a-fe39-47bc-9c18-0c23fa3cec65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA3MzM0NzYsIm5iZiI6MTc0MDczMzE3NiwicGF0aCI6Ii81NDg0NzI2NC8zNzczNTE3MDItMGNhYWU4NGEtZmUzOS00N2JjLTljMTgtMGMyM2ZhM2NlYzY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI4VDA4NTkzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhOWFiYmU2YzNjYWRmN2FlNjFmMTMzZTRkZDc5NGNjNDI1ZDg0MTRkNTAxNzQzOWYzMzcxM2IwYzQ5ZGY0NWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MYlftbvq26xV6Nz9_y1vHUDtvNBdoUQzl22ztqB1Iso) 你好，你是开发者吗？在使用停止命令遇到一些错误，执行停止命令不能正常停止服务，这种情况发生在最近任务没有cmfa窗口

我遇到了更严重的问题，除了关闭服务不管用以外，如果最近任务里有cmfa窗口，那么发送开启或者关闭服务的intent都会立刻弹进cmfa应用，而我在macrodroid里设置的是打开一些app时自动发送intent，所以此时如果尝试切回原本的应用时又会立刻给我弹回cmfa……这个问题不知道什么时候能够解决，不解决的话感觉完全没法用的样子……

**Comment by cplasfwst** [comment]
请问是否能通过命令行导入Profile文件？请求指点
